src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -85,29 +93,40 @@ public StorageFile(String filePath) throws InvalidStorageFilePathException {
     private static boolean isValidPath(Path filePath) {
         return filePath.toString().endsWith(".xml");
     }
+    
+    /**
+     * Check if the file from the given file path exist
+     * @return true if file exist and false otherwise
+     */
+    private boolean isFileExist() {
+        File file = path.toFile();
+        return file.exists();
+    }
 
     /**
      * Saves all data to this storage file.
      *
+     * @throws FileNotFoundException if the file does not exist.
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+        if (!isFileExist()) {
```

Comment:
you have got the basic idea right; however, in the current form, the user experience is degraded.
as soon as the file goes missing, the program halts.

in the original implementation, the file is created on the fly; think how you can retain the behavior while still using exception as a means to inform the user that the file went missing!!